packages/compose/src/hooks/use-constrained-tabbing/index.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * WordPress dependencies
+ */
+import { useCallback } from '@wordpress/element';
+import { TAB } from '@wordpress/keycodes';
+import { focus } from '@wordpress/dom';
+
+/**
+ * In Dialogs/modals, the tabbing must be constrained to the content of
+ * the wrapper element. This hook adds the behavior to the returned ref.
+ *
+ * @return {Function} Element Ref.
+ *
+ * @example
+ * ```js
+ * import { useConstrainedTabbing } from '@wordpress/compose';
+ *
+ * const ConstrainedTabbingExample = () => {
+ *     const constrainedTabbingRef = useConstrainedTabbing()
+ *     return (
+ *         <div ref={ constrainedTabbingRef }>
+ *             <Button />
+ *             <Button />
+ *         </div>
+ *     );
+ * }
+ * ```
+ */
+function useConstrainedTabbing() {
+	const ref = useCallback( ( node ) => {
```

Comment:
Why not add the listener on `useEffect`? That way the ref is also available for other things.

Reply:
So the API would be `useConstrainedTabbing( ref )` and the implementor creates the `ref` with `useRef`.

Reply:
I commented about this in another PR. Passing a ref is a bad pattern because it doesn't update when the "ref" changes. This is a better pattern for the hooks that require refs. Create the ref inside the hook and return it and if someone wants to use multiple refs, they can use `mergeRefs`

Reply:
Ok, let's update other instances then